core/services/ocr2/plugins/functions/config/config.go

```diff
@@ -101,3 +104,29 @@ func (ThresholdConfigParser) ParseConfig(config []byte) (*decryptionPluginConfig
 		},
 	}, nil
 }
+
+func S4ConfigDecoder(config []byte) (*s4PluginConfig.PluginConfig, *types.ReportingPluginLimits, error) {
+	reportingPluginConfigWrapper, err := DecodeReportingPluginConfig(config)
+	if err != nil {
+		return nil, nil, errors.New("failed to decode S4 plugin config")
+	}
+
+	pluginConfig := reportingPluginConfigWrapper.Config.S4PluginConfig
+	if pluginConfig == nil {
+		return nil, nil, fmt.Errorf("PluginConfig bytes %x did not contain s4 plugin config", config)
```

Comment:
Does it make sense to log all the raw bytes?

Reply:
I believe this was copied from another plugin. Maybe we don't need to log it, because we always have it on-chain?